NewStarterProject/Services/UserService.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Microsoft.EntityFrameworkCore;
+using NewStarterProject.Dtos;
+using NewStarterProject.Model;
+
+namespace NewStarterProject.Services
+{
+    public class UserService
+    {
+        
+        private StarterProjectContext _context;
+
+        public UserService(StarterProjectContext context)
+        {
+            _context = context;
+        }
+
+
+        /// <summary>
+        /// Create User
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public async Task<UserDto> CreateOrUpdateUser(UserDto userDto)
+        {
+            User user;
+
+            // If the user already exists (has an Id), get it from the Db
+            if (userDto.UserId > 0)
+            {
+                user = await _context.Users.SingleAsync(a => a.UserId == userDto.UserId);
+            }
+            else
+            {
+                user = new User
+                {
+                    IsActive = true
+                };
+                _context.Users.Add(user);
+            }
+
+            // Update the values - obviously these will be empty for a new user
+
+            if (user.FirstName != userDto.FirstName)
+            {
+                user.FirstName = userDto.FirstName;
+            }
+
+            if (user.LastName != userDto.LastName)
+            {
+                user.LastName = userDto.LastName;
+            }
+
+            if (user.UserName != userDto.UserName)
+            {
+                user.UserName = userDto.UserName;
+            }
+
+            if (user.PinCode != userDto.PinCode)
+            {
+                user.PinCode = userDto.PinCode;
+            }
+
+            await _context.SaveChangesAsync();
+            userDto.UserId = userDto.UserId;
+
+            return userDto;
+
+        }
+
+        /// <summary>
+        /// Get a User by Id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public async Task<UserDto> Get(int id)
+        {
+            var user = await _context.Users.SingleOrDefaultAsync(a => a.UserId == id && a.IsActive);
+
+            return new UserDto
+            {
+                UserId = user?.UserId ?? -1,
+                FirstName = user.FirstName ?? "",
+                LastName = user.LastName ?? "",
+                PinCode = user.PinCode ?? "",
+                UserName = user.UserName ?? ""
+            };
```

Comment:
Nice null handling here with the null coalescing operators (??) we had bugs in the past because we did not handle nulls like this so well done 😎